File: ros2_ws/src/my_cpp_pkg/src/target_coordinates_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "custom_interfaces1/msg/TargetCoordinates.hpp"

class TargetCoordinatesPublisher : public rclcpp::Node 
{
public:
    TargetCoordinatesPublisher() : Node("robot_news_station")
    {
        publisher_ = this->create_publisher<example_interfaces::msg::String>("target_coordinates",10);
        timer_ = this->create_wall_timer(std::chrono::milliseconds(500), std::bind(&TargetCoordinatesPublisher::publishTargetCoordinates, this));
        RCLCPP_INFO(this->get_logger(), "Robot Targer Coordinates has been started.");
    }

private:
    void publishTargetCoordinates()
    {
        auto msg = custom_interfaces1::msg::TargetCoordinates();
        msg.name = "robottt"
            
        geometry_msgs::Point point;
        point.x = 1.0;
        point.y = 2.0;
        point.z = 3.0;

    
        ROS_INFO("Point: x = %f, y = %f, z = %f", point.x, point.y, point.z);
             publisher_->publish(msg);
          }

    std::string robot_name_;
    rclcpp::Publisher<custom_interfaces1::msg::TargetCoordinates>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<TargetCoordinatesPublisher>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: ros2_ws/src/my_py_pkg/my_py_pkg/target_coordinates_publisher.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from custom_interfaces1.msg import TargetCoordinates
from geometry_msgs.msg import Point

class TargetCoordinatesPublisher(Node):

    def __init__(self):
        super().__init__("target_coordinates_publisher")
        self.coordinates_publisher_ = self.create_publisher(TargetCoordinates, "target_coordinates", 10)
        self.timer_ = self.create_timer(1.0, self.publish_target_coordinates)
        self.get_logger().info("Coordinates publisher has been started.")

    def publish_target_coordinates(self):
        msg = TargetCoordinates()
        msg.name = "Robot"
        point = Point()
        point.x = 1.0  
        point.y = 2.0  
        point.z = 3.0  
        msg.coordinates = point
        self.coordinates_publisher_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = TargetCoordinatesPublisher()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()
